MoneyDataType/KnownCurrencyTable.cs

```diff
@@ -1,49 +1,53 @@
+    internal static IDictionary<string, ICurrency> CurrencyTable { get; private set; }
 
-        internal static void EnsureCurrencyTable()
-        {
-            if (CurrencyTable == null)
-                InitCurrencyCodeTable();
-        }
+    internal static void EnsureCurrencyTable()
+    {
+        if (CurrencyTable == null) InitCurrencyCodeTable();
+    }
 
-        private class CurrencyEqualityComparer : IEqualityComparer<ICurrency>
-        {
-            public bool Equals(ICurrency x, ICurrency y) => x.CurrencyIsoCode == y.CurrencyIsoCode;
-            public int GetHashCode(ICurrency obj) => obj.CurrencyIsoCode.GetHashCode();
-        }
+    private sealed class CurrencyEqualityComparer : IEqualityComparer<ICurrency>
+    {
+        public bool Equals(ICurrency left, ICurrency right) =>
+            (left is null && right is null) ||
+            left?.CurrencyIsoCode?.Equals(right?.CurrencyIsoCode, StringComparison.OrdinalIgnoreCase) == true;
+        public int GetHashCode(ICurrency obj) => StringComparer.OrdinalIgnoreCase.GetHashCode(obj.CurrencyIsoCode);
+    }
 
-        private static void InitCurrencyCodeTable()
-        {
-            lock (Obj)
-            {
-                bool valid(CultureInfo c) => !c.IsNeutralCulture && !c.EnglishName.StartsWith("Unknown Locale") && !c.EnglishName.StartsWith("Invariant Language");
-
-                CurrencyTable = CultureInfo.GetCultures(CultureTypes.AllCultures)
-                    .Where(valid)
-                    .Select(c => new Currency(c)).Cast<ICurrency>()
-                    .Distinct(new CurrencyEqualityComparer())
-                    .ToDictionary(k => k.CurrencyIsoCode, e => e);
-
-                CurrencyTable.Add("BTC", new Currency("BitCoin", "BitCoin", "₿", "BTC", 8));
-                CurrencyTable.Add("---", new Currency("Unspecified", "Unspecified", "---", "---"));
-            }
-        }
+    private static void InitCurrencyCodeTable()
+    {
+        static bool Valid(CultureInfo cultureInfo) =>
+            !cultureInfo.IsNeutralCulture &&
+            !cultureInfo.EnglishName.StartsWith("Unknown Locale", StringComparison.Ordinal) &&
+            !cultureInfo.EnglishName.StartsWith("Invariant Language", StringComparison.Ordinal);
 
-        internal static ICurrency FromIsoCode(string isoCode)
+        lock (_obj)
         {
-            EnsureCurrencyTable();
-            return CurrencyTable[isoCode];
+            CurrencyTable = CultureInfo.GetCultures(CultureTypes.AllCultures)
+                .Where(Valid)
+                .Select(c => new Currency(c)).Cast<ICurrency>()
+                .Distinct(new CurrencyEqualityComparer())
+                .ToDictionary(k => k.CurrencyIsoCode, e => e);
+
+            CurrencyTable.Add("BTC", new Currency("BitCoin", "BitCoin", "₿", "BTC", 8));
+            CurrencyTable.Add("---", new Currency("Unspecified", "Unspecified", "---", "---"));
```

Comment:
Use the predefined `UnspecifiedCurrency` here:
```suggestion
            CurrencyTable.Add("---", Currency.UnspecifiedCurrency);
```